pkg/apis/monitoring/v1/types.go

```diff
@@ -1093,8 +1093,12 @@ type ServiceMonitor struct {
 // ServiceMonitorSpec contains specification parameters for a ServiceMonitor.
 // +k8s:openapi-gen=true
 type ServiceMonitorSpec struct {
-	// Chooses the label of the Kubernetes `Endpoints`.
-	// Its value will be used for the `job`-label's value of the created metrics.
+	// JobLabel chooses the label of the Kubernetes `Endpoints`.
+	// The value of this field will be used as the value for the `job`-label in each created metric.
+	//
+	// For example:
+	// If in `ServiceMonitor.spec.jobLabel: foo` and in `Service.metadata.labels.foo: bar`°,
+	// then each metric gets `{job="bar"}`-label. (° to be precise: in the `Endpoint`)
```

Comment:
IIUC the Prometheus config looks at the service's labels so I wouldn't mention endpoints. 

```suggestion
	// If in `ServiceMonitor.spec.jobLabel: foo` and in `Service.metadata.labels.foo: bar`,
	// then each metric gets `{job="bar"}`-label.
```

Reply:
@simonpasquier Yes, but AFAICS the `prometheus-operator` sets the `role: endpoints` in `func (cg *ConfigGenerator) generateServiceMonitorConfig(...)`

https://github.com/prometheus-operator/prometheus-operator/blob/f1e140a982400e7019689cef7aad93d339e64884/pkg/prometheus/promcfg.go#L1165

Otherwise the [comment below](https://github.com/prometheus-operator/prometheus-operator/pull/4862/files#diff-255c0aca616a1cdf4385c88d1d063a6c86360fedf99f7fc76bb967034e68897cR1103) must be updated:
```golang
// Default & fallback value: the name of the respective Kubernetes `Endpoint`.
```

Reply:
I think the next sentence should be updated indeed. The relevant code is:

https://github.com/prometheus-operator/prometheus-operator/blob/f1e140a982400e7019689cef7aad93d339e64884/pkg/prometheus/promcfg.go#L1356-L1371

Reply:
Ah right, thx. 👍